### PR TITLE
PYTHON-4594 - Add to_list documentation

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: pymongo/asynchronous/command_cursor.py [hunk 1]
```diff
@@ -382,6 +382,12 @@ async def __aexit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
         await self.close()
 
     async def to_list(self) -> list[_DocumentType]:
+        """Converts the contents of this cursor to a list.
+
+        If the cursor is empty or has no more results, an empty list will be returned.
```
Comment (ShaneHarvey) [inline pymongo/asynchronous/command_cursor.py:387]: Suggest adding an example and saying this is a more efficient verison of `[doc async for doc in cursor]`.

Comment (NoahStapp) [reply pymongo/asynchronous/command_cursor.py:391]: Done. The synchronization to the sync versions will be handled by https://github.com/mongodb/mongo-python-driver/pull/1756.

File: pymongo/synchronous/command_cursor.py [hunk 2]
```diff
@@ -382,6 +382,16 @@ def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
         self.close()
 
     def to_list(self) -> list[_DocumentType]:
+        """Converts the contents of this cursor to a list more efficiently than ``[doc async for doc in cursor]``.
```
Comment (Jibola) [inline pymongo/synchronous/command_cursor.py:385]:
```suggestion
        """Converts the contents of this cursor to a list more efficiently than ``[doc for doc in cursor]``.
```

I don't think we need to supply that language here. That or we just say above.

File: pymongo/synchronous/cursor.py [hunk 3]
```diff
@@ -1285,6 +1285,16 @@ def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
         self.close()
 
     def to_list(self) -> list[_DocumentType]:
+        """Converts the contents of this cursor to a list more efficiently than ``[doc async for doc in cursor]``.
```
Comment (Jibola) [inline pymongo/synchronous/cursor.py:1288]:
```suggestion
        """Converts the contents of this cursor to a list more efficiently than ``[doc for doc in cursor]``.
```

**Review by Jibola — CHANGES_REQUESTED** [review]

**Review by ShaneHarvey — APPROVED** [review]

File: pymongo/synchronous/command_cursor.py (hunk 2 above)
Comment (NoahStapp) [reply pymongo/synchronous/command_cursor.py:385]: Sorry, which language?

File: pymongo/synchronous/cursor.py (hunk 3 above)
Comment (NoahStapp) [reply pymongo/synchronous/cursor.py:1288]: https://github.com/mongodb/mongo-python-driver/pull/1756 will fix the inline examples not being synchronized.

File: pymongo/synchronous/command_cursor.py (hunk 2 above)
Comment (Jibola) [reply pymongo/synchronous/command_cursor.py:385]: The `async` part. I see you addressed it earlier.

**Review by Jibola — APPROVED** [review]

**Pull request merged**